src/transformStateWithClones.js

```
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let temp = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    temp = doAction(temp, action);
+    result.push({ ...temp });
+  }
+
+  return result;
+}
+
+function doAction(state, action) {
+  switch (action.type) {
+    case 'addProperties':
+      for (const key in action.extraData) {
+        state[key] = action.extraData[key];
+      }
+
+      return state;
+
+    case 'removeProperties':
+      for (const key of action.keysToRemove) {
+        delete state[key];
+      }
+
+      return state;
+
+    case 'clear':
+      for (const key in state) {
+        delete state[key];
+      }
+
+      return state;
+  }
```

This violates checklist item #3: "3. [CODE STYLE]: switch/case should always have default case for error handling." The `switch` in `doAction` has no `default` branch. Without a default, an unknown `action.type` will make `doAction` return `undefined`, which will set `temp` to `undefined` and then `result.push({ ...temp })` will throw. Add a `default` case that either throws a clear error or returns the unchanged `state` so the function fails fast or continues safely. See the checklist guidance: switch/case should always have default case for error handling.